File: test/data/call_extraction/calls.go
package main

import "fmt"

func helper() int {
	return 42
}

func process(data string) string {
	result := transform(data)
	validated := validate(result)
	fmt.Println("processing")
	return validated
}

func transform(data string) string {
	return data
}

func validate(data string) bool {
	return len(data) > 0
}

type Service struct {
	db string
}

func NewService(db string) *Service {
	return &Service{db: db}
}

func (s *Service) Fetch(query string) []string {
	return s.db.Execute(query)
}

func (s *Service) ProcessAll() []string {
	data := s.Fetch("SELECT 1")
	return transform(data[0])
}

func outer() {
	helper()
	process("test")
}

func main() {
	svc := NewService("db")
	fmt.Println(svc)
}

File: test/data/call_extraction/calls.rs
fn helper() -> i32 {
    42
}

fn process(data: &str) -> String {
    let result = transform(data);
    let validated = validate(&result);
    println!("processing");
    validated
}

fn transform(data: &str) -> String {
    data.to_string()
}

fn validate(data: &str) -> String {
    data.trim().to_string()
}

struct Service {
    db: String,
}

impl Service {
    fn new(db: &str) -> Self {
        Service { db: db.to_string() }
    }

    fn fetch(&self, query: &str) -> Vec<String> {
        self.db.execute(query)
    }

    fn process_all(&self) -> Vec<String> {
        let data = self.fetch("SELECT 1");
        transform(&data[0])
    }
}

fn outer() {
    fn inner() {
        helper();
    }
    inner();
    process("test");
}

fn main() {
    let svc = Service::new("db");
    println!("done");
}
